src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/service/impl/ValidationServiceImpl.java

```diff
@@ -200,6 +210,54 @@ public ValidateGroupsResult validateGroups(SwitchId switchId, List<GroupEntry> g
                 ImmutableList.copyOf(misconfiguredGroups));
     }
 
+    @Override
+    public ValidateLogicalPortsResult validateLogicalPorts(SwitchId switchId, List<LogicalPort> presentLogicalPorts) {
+        Map<Integer, LogicalPortInfoEntry> expectedPorts = lagLogicalPortRepository.findBySwitchId(switchId).stream()
+                .map(LogicalPortMapper.INSTANCE::map)
+                .collect(Collectors.toMap(LogicalPortInfoEntry::getLogicalPortNumber, Function.identity()));
+
+        Map<Integer, LogicalPortInfoEntry> actualPorts = presentLogicalPorts.stream()
+                .map(LogicalPortMapper.INSTANCE::map)
+                .collect(Collectors.toMap(LogicalPortInfoEntry::getLogicalPortNumber, Function.identity()));
+
+        List<LogicalPortInfoEntry> properPorts = new ArrayList<>();
+        List<LogicalPortInfoEntry> missingPorts = new ArrayList<>();
+        List<LogicalPortInfoEntry> excessPorts = new ArrayList<>();
+        List<LogicalPortInfoEntry> misconfiguredPorts = new ArrayList<>();
+
+        for (Entry<Integer, LogicalPortInfoEntry> entry : expectedPorts.entrySet()) {
+            int portNumber = entry.getKey();
+            LogicalPortInfoEntry expected = entry.getValue();
+
+            if (actualPorts.containsKey(portNumber)) {
+                LogicalPortInfoEntry actual = actualPorts.get(portNumber);
```

Comment:
You can use `.remove()` here. So everithing that left into `actualPorts` after this loop is `excess` (except BFD).